webapp/tests/forms/steps/test_eligibility_steps.py

```diff
@@ -0,0 +1,310 @@
+                eligibility_step._main_handle(self.data_not_eligible)
+
+                self.assertEqual(not_eligible_errors, eligibility_step.render_info.additional_info['eligibility_errors'])
+
+    def test_if_keys_not_in_data_and_income_step_then_return_422(self):
+        with app.app_context() and app.test_request_context(method='GET'):
+            eligibility_step = EligibilityResultFormSteuerlotseStep(endpoint=self.endpoint_correct, next_step=EligibilityResultFormSteuerlotseStep)
+            eligibility_step._pre_handle(self.data_without_all_keys)
+
+            self.assertRaises(IncorrectEligibilityData,
+                              eligibility_step._main_handle, self.data_without_all_keys)
+
+
+class TestEligibilityIncomesSteuerlotseStepHandle(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.endpoint_correct = "eligibility"
+        self.data_not_eligible = {'renten': 'no', 'kapitaleink_mit_steuerabzug': 'yes',
+                                  'kapitaleink_ohne_steuerabzug': 'no', 'kapitaleink_mit_pauschalbetrag': 'yes',
+                                  'kapitaleink_guenstiger': 'no', 'geringf': 'yes', 'erwerbstaetigkeit': 'yes',
+                                  'unterhalt': 'yes', 'ausland': 'no', 'other': 'no',
+                                  'verheiratet_zusammenveranlagung': 'yes', 'verheiratet_einzelveranlagung': 'no',
+                                  'geschieden_zusammenveranlagung': 'no', 'elster_account': 'no'}
+        self.data_eligible = {'renten': 'yes', 'kapitaleink_mit_steuerabzug': 'yes',
+                              'kapitaleink_ohne_steuerabzug': 'no', 'kapitaleink_mit_pauschalbetrag': 'yes',
+                              'kapitaleink_guenstiger': 'no', 'geringf': 'yes', 'erwerbstaetigkeit': 'no',
+                              'unterhalt': 'no', 'ausland': 'no', 'other': 'no',
+                              'verheiratet_zusammenveranlagung': 'yes', 'verheiratet_einzelveranlagung': 'no',
+                              'geschieden_zusammenveranlagung': 'no', 'elster_account': 'no'}
+        self.result_url = '/' + self.endpoint_correct + '/step/' + MockEligibilityResultFormStep.name + \
+                              '?link_overview='
+
+    def test_if_income_step_then_set_next_url_correct(self):
+        with patch("app.forms.steps.eligibility_steps._") as babel:
+            babel.side_effect = lambda arg: arg
+
+            with app.app_context() and app.test_request_context():
+                eligibility_step = EligibilityIncomesFormSteuerlotseStep(endpoint=self.endpoint_correct, next_step=EligibilityResultFormSteuerlotseStep)
+                eligibility_step._pre_handle(self.data_not_eligible)
+                eligibility_step._main_handle(self.data_not_eligible)
+
+                self.assertEqual(self.result_url + str(eligibility_step.has_link_overview), eligibility_step.render_info.next_url)
+
+    def test_if_income_step_then_set_no_additional_info(self):
```

Comment:
I've removed that test completely, because we are testing that something is not happening which we do not do. I think this does not make sense any longer